src/v/cloud_storage/tests/remote_segment_test.cc

```diff
@@ -64,11 +64,11 @@ static cloud_storage::lazy_abort_source always_continue("no-op", [](auto&) {
  * exposing some synthetic data as a segment_reader_handle.
  */
 remote::reset_input_stream make_reset_fn(iobuf& segment_bytes) {
-    return [&segment_bytes]()
-             -> ss::future<std::unique_ptr<storage::stream_provider>> {
+    return [&segment_bytes] {
```

Comment:
The conversion is good, but the pattern is concerning. This is used inside a threaded test and lifetimes seem ok.

Reply:
what's the concern?

Reply:
returning a lambda with a captured reference is hard to reason about without looking at the whole stack, but in this case it's short and ok.

Reply:
Oh, right. That's what I meant by "The conversion is good,".